File: 7.rb
def prime?(n)
	(2..n/2).each{ |x| return false if n % x == 0 }
	true
end

def nth_prime(n)
	count = 2
	i = 4
	while true
		count += 1 if prime?(i)
		break if count == n
		i += 1
	end
	return i
end

puts nth_prime(10001)

File: 7.py
def isPrime(n):
    if n <= 1: return False
    for i in xrange(2, n // 2 + 1):
        if n % i == 0:
            return False
    return True

count = 0
x = 0
while count < 10001:
    x += 1
    if isPrime(x): count += 1

print x